devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -992,8 +1011,69 @@ void YarpRobotLoggerDevice::lookForExogenousSignals()
         wakeUpTime += lookForExogenousSignalPeriod;
 
         // try to connect to the exogenous signals
-        connectToExogeneous(m_vectorsCollectionSignals);
-        connectToExogeneous(m_vectorSignals);
+        for (auto& [name, signal] : m_vectorsCollectionSignals)
+        {
+            std::lock_guard<std::mutex> lock(signal.mutex);
+
+            // the connection is established
+            if (signal.signal.connected && signal.metadataClient.connected)
+            {
+                continue;
+            }
+
+            // try to connect the signal
+            if (!signal.signal.connected && yarp::os::Network::exists(signal.signal.remote))
```

Comment:
Is the ``exists`` really necessary? The subsequent ``connect`` would fail anyhow. I guess that you are saving some prints in the terminal, but it could take longer to start.